MediaBrowser.Controller/MediaEncoding/DownMixAlgorithmsHelper.cs

```diff
@@ -0,0 +1,61 @@
+/// </summary>
+public static class DownMixAlgorithmsHelper
+{
+    /// <summary>
+    /// The filter string of the DownMixStereoAlgorithms.
+    /// The index is the tuple of (algorithm, layout).
+    /// </summary>
+    public static readonly Dictionary<(DownMixStereoAlgorithms, string), string> AlgorithmFilterStrings = new()
+    {
+        { (DownMixStereoAlgorithms.Dave750, "5.1"), "pan=stereo|c0=0.5*c2+0.707*c0+0.707*c4+0.5*c3|c1=0.5*c2+0.707*c1+0.707*c5+0.5*c3" },
+        { (DownMixStereoAlgorithms.NightmodeDialogue, "5.1"), "pan=stereo|c0=c2+0.30*c0+0.30*c4|c1=c2+0.30*c1+0.30*c5" },
+        { (DownMixStereoAlgorithms.Rfc7845, "3.0"), "pan=stereo|c0=0.414214*c2+0.585786*c0|c1=0.414214*c2+0.585786*c1" },
+        { (DownMixStereoAlgorithms.Rfc7845, "quad"), "pan=stereo|c0=0.422650*c0+0.366025*c2+0.211325*c3|c1=0.422650*c1+0.366025*c3+0.211325*c2" },
+        { (DownMixStereoAlgorithms.Rfc7845, "5.0"), "pan=stereo|c0=0.460186*c2+0.650802*c0+0.563611*c3+0.325401*c4|c1=0.460186*c2+0.650802*c1+0.563611*c4+0.325401*c3" },
+        { (DownMixStereoAlgorithms.Rfc7845, "5.1"), "pan=stereo|c0=0.374107*c2+0.529067*c0+0.458186*c4+0.264534*c5+0.374107*c3|c1=0.374107*c2+0.529067*c1+0.458186*c5+0.264534*c4+0.374107*c3" },
+        { (DownMixStereoAlgorithms.Rfc7845, "6.1"), "pan=stereo|c0=0.321953*c2+0.455310*c0+0.394310*c5+0.227655*c6+0.278819*c4+0.321953*c3|c1=0.321953*c2+0.455310*c1+0.394310*c6+0.227655*c5+0.278819*c4+0.321953*c3" },
+        { (DownMixStereoAlgorithms.Rfc7845, "7.1"), "pan=stereo|c0=0.274804*c2+0.388631*c0+0.336565*c6+0.194316*c7+0.336565*c4+0.194316*c5+0.274804*c3|c1=0.274804*c2+0.388631*c1+0.336565*c7+0.194316*c6+0.336565*c5+0.194316*c4+0.274804*c3" },
+    };
+
+    /// <summary>
+    /// Get the audio channel layout string from the audio stream
+    /// If the input audio string does not have a valid layout string, guess from channel count.
+    /// </summary>
+    /// <param name="audioStream">The audio stream to get layout.</param>
+    /// <returns>Channel Layout string.</returns>
+    public static string InferChannelLayout(MediaStream audioStream)
+    {
+        if (!string.IsNullOrWhiteSpace(audioStream.ChannelLayout))
+        {
+            return audioStream.ChannelLayout;
+        }
+
+        if (audioStream.Channels is null)
+        {
+            return string.Empty;
+        }
+
+        // When we don't have definitive channel layout, we have to guess from the channel count
+        // Guessing is not always correct, but for most videos we don't have to guess like this as the definitive layout is recorded during scan
+        var inferredLayout = audioStream.Channels.Value switch
+        {
+            1 => "mono",
+            2 => "stereo",
+            3 => "2.1", // Could also be 3.0, prefer 2.1
+            4 => "quad", // Could also be 4.0 (with front center and rear center) and 3.1 with LFE. prefer quad with back left and back right
+            5 => "5.0",
```

Comment:
Why not prefer 4.1?

Reply:
Because 5.0 is more common? But this is purely subjective and such guessing could be wrong anyway.

Reply:
I've personally never seen 5.0, but have seen 4.1
https://www.blu-ray.com/movies/Alien-4K-Blu-ray/230308/

Reply:
We just fixed an issued related to 5.0 in 10.9: https://github.com/jellyfin/jellyfin/issues/11397

Reply:
Search with [4.1 channel resulted in 37 matches](https://www.blu-ray.com/movies/search.php?keyword=&studioid=&videocodec=&disc=&yearfrom=&yearto=&regioncoding=&aspectratio=&aspectratio_original=&releaseyear=&synopsis=&retailerexclusive=&mpaa=&runtimemin=&runtimemax=&videoresolutionid=&sourceid=&subtitles=&audio=4.1&upc=&ean=&asin=&casingid=&slipcoverfront=&slipcoverback=&submit=Search&action=search) and [5.0 resulted in 303 matches](https://www.blu-ray.com/movies/search.php?keyword=&studioid=&videocodec=&disc=&yearfrom=&yearto=&regioncoding=&aspectratio=&aspectratio_original=&releaseyear=&synopsis=&retailerexclusive=&mpaa=&runtimemin=&runtimemax=&videoresolutionid=&sourceid=&subtitles=&audio=5.0&upc=&ean=&asin=&casingid=&slipcoverfront=&slipcoverback=&submit=Search&action=search). I think we should stay with 5.0.